pkg/sources/github/github.go

```diff
@@ -772,50 +780,35 @@ func (s *Source) scan(ctx context.Context, installationClient *github.Client, ch
 				return nil
 			}
 
-			logger := s.log.WithValues("repo", repoURL)
-			logger.V(2).Info("attempting to clone repo")
-			var (
-				path string
-				repo *gogit.Repository
-				err  error
-			)
-
-			path, repo, err = s.cloneRepo(ctx, repoURL, installationClient)
+			// Scan the repository
+			repoCtx := context.WithValues(ctx, "repo", repoURL)
+			duration, err := s.cloneAndScanRepo(repoCtx, installationClient, repoURL, chunksChan)
 			if err != nil {
 				scanErrs.Add(err)
 				return nil
 			}
-
-			defer os.RemoveAll(path)
-			if err != nil {
-				scanErrs.Add(fmt.Errorf("error cloning repo %s: %w", repoURL, err))
-				return nil
-			}
-
-			s.setScanOptions(s.conn.Base, s.conn.Head)
-
-			repoSize := s.repoSizes.getRepo(repoURL)
-			logger.V(2).Info("scanning repo", "repo_size_kb", repoSize)
-
-			now := time.Now()
-			scanFailed := false
-			defer func(start time.Time, failed *bool) {
-				if *failed {
-					return
+			ctx.Logger().V(2).Info(fmt.Sprintf("scanned %d/%d repos", scannedCount, len(s.repos)), "duration_seconds", duration)
+
+			// Scan the wiki, if enabled, and the repo has one.
+			if s.conn.IncludeWikis {
+				if _, ok := s.reposWithWikis[repoURL]; ok {
+					wikiURL := strings.TrimSuffix(repoURL, ".git") + ".wiki.git"
+					wikiCtx := context.WithValue(ctx, "repo", wikiURL)
+
+					_, err := s.cloneAndScanRepo(wikiCtx, installationClient, wikiURL, chunksChan)
+					if err != nil {
+						scanErrs.Add(err)
+						// Don't return, it still might be possible to scan comments.
+					}
 				}
-				logger.V(2).Info(fmt.Sprintf("scanned %d/%d repos", scannedCount, len(s.repos)), "duration_seconds", time.Since(start).Seconds())
-			}(now, &scanFailed)
-
-			if err = s.git.ScanRepo(ctx, repo, path, s.scanOptions, sources.ChanReporter{Ch: chunksChan}); err != nil {
-				scanErrs.Add(fmt.Errorf("error scanning repo %s: %w", repoURL, err))
-				scanFailed = true
-				return nil
 			}
 
-			if err = s.scanComments(ctx, repoURL, chunksChan); err != nil {
-				scanErrs.Add(fmt.Errorf("error scanning comments in repo %s: %w", repoURL, err))
-				scanFailed = true
-				return nil
+			// Scan comments, if enabled.
+			if s.includeGistComments || s.includeIssueComments || s.includePRComments {
```

Comment:
I added an explicit check because I think it's more consistent and readable to have `if includeWiki { doWiki } ... if includeComments { doComments }`.

The `s.includeGistComments`, `s.includeIssueComments`, and `s.includePRComments` weren't being _ignored_, but they were implemented in a way that's awkward and difficult to understand (imo). I also don't think it makes sense to enter a function that can return errors when the flags are disabled.

https://github.com/trufflesecurity/trufflehog/blob/9867ce8eb8366405f4ff97670a897da6139966c7/pkg/sources/github/github.go#L1104-L1107

https://github.com/trufflesecurity/trufflehog/blob/9867ce8eb8366405f4ff97670a897da6139966c7/pkg/sources/github/github.go#L1179-L1182

Reply:
Oof, strong agree. Thanks for this.